pyiron_base/project/archiving/export_archive.py

```diff
@@ -78,36 +59,21 @@ def copy_files_to_archive(
 
     assert isinstance(archive_directory, str) and ".tar.gz" not in archive_directory
     # print("directory to transfer: "+directory_to_transfer)
-    if not copy_all_files:
-        pfi = PyFileIndex(path=directory_to_transfer, filter_function=filter_function)
+    dst = os.path.join(archive_directory, getdir(path=directory_to_transfer))
+    if copy_all_files:
+        copytree(directory_to_transfer, dst, dirs_exist_ok=True)
     else:
-        pfi = PyFileIndex(path=directory_to_transfer)
-    df_files = pfi.dataframe[~pfi.dataframe.is_directory]
-
-    # Create directories
-    dir_lst = generate_list_of_directories(
-        df_files=df_files,
-        directory_to_transfer=directory_to_transfer,
-        archive_directory=archive_directory,
-    )
-    # print(dir_lst)
-    for d in dir_lst:
-        os.makedirs(d, exist_ok=True)
-    # Copy files
-    dir_name_transfer = getdir(path=directory_to_transfer)
-    for f in df_files.path.values:
-        copyfile(
-            f,
-            os.path.join(
-                archive_directory,
-                dir_name_transfer,
-                os.path.relpath(f, directory_to_transfer),
-            ),
+        copytree(
+            directory_to_transfer, dst, ignore=ignore_non_h5_files, dirs_exist_ok=True
         )
     if compressed:
         compress_dir(archive_directory)
 
 
+def ignore_non_h5_files(dir, files):
```

Comment:
Could be replaced with `shutil.ignore_patterns('*.h5')`, but aesthetic only.

Does this work with nested projects/child jobs? If I understand the docs correctly `copytree` passes the files and directories and if you "ignore" directories does not recurse into them, but I may just misunderstand.

Reply:
It's actually other way around: Only files ending with `.h5` are not to be ignored.

Reply:
> Does this work with nested projects/child jobs? If I understand the docs correctly `copytree` passes the files and directories and if you "ignore" directories does not recurse into them, but I may just misunderstand.

I got a reply from ChatGPT:

### Example Directory Structure

Suppose you have the following directory structure:

```
source_directory/
|-- file1.txt
|-- file2.h5
|-- file3.py
|-- subdir/
    |-- file4.h5
    |-- file5.txt
```

After running the script, `destination_directory` will have the following structure:

```
destination_directory/
|-- file2.h5
|-- subdir/
    |-- file4.h5
```

Reply:
> > Could be replaced with shutil.ignore_patterns('*.h5'), but aesthetic only.
> 
> It's actually other way around: Only files ending with `.h5` are not to be ignored.

Ah, I read too quickly.